### PR TITLE
passwordless ssh

### PR DESCRIPTION
**Pull request by smruti77**
removing running command through ssh on peer machine and adding
login to the peer machine and running the command through function.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

File: io/net/bonding.py [hunk 1]
```diff
@@ -27,6 +27,11 @@
 import socket
 import fcntl
 import struct
+try:
+    import pxssh
+except ImportError:
+    from pexpect import pxssh
+import netifaces
```
Comment (narasimhan-v) [inline io/net/bonding.py]: imported netifaces twice

File: io/net/bonding.py [hunk 2]
```diff
@@ -90,9 +108,47 @@ def setUp(self):
...
+            self.fail("SSH Console setup is not yet done")
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = con.before.splitlines()
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        exitcode = int(''.join(con.before.splitlines()[1:]))
+        return (output, exitcode)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: please check if this functions (mainly return values) are same across all the tests

File: io/net/bonding.py [hunk 3]
```diff
@@ -68,6 +84,8 @@ def setUp(self):
                 self.cancel("test skipped because mode not specified")
         interfaces = netifaces.interfaces()
         self.user = self.params.get("user_name", default="root")
+        self.password = self.params.get("peer_password", '*',
+                                        default=None)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: please default it to `passw0rd` either in yaml or here.

File: io/net/bonding.py.data/bonding.yaml [hunk 4]
```diff
@@ -16,6 +16,7 @@ Test: !mux
 bond_interfaces: ""
 peer_ip: ""
 peer_interfaces: ""
+peer_password: ""
```
Comment (narasimhan-v) [inline io/net/bonding.py.data/bonding.yaml]: Check in readme file about mention of passwordless ssh setup, and change accordingly.
This is for all tests.

File: io/net/iperf_test.py [hunk 5]
```diff
@@ -111,9 +169,10 @@ def tearDown(self):
         """
         Killing Iperf process in peer machine
         """
-        cmd = "ssh %s@%s \"pkill iperf; rm -rf /tmp/iperf-master\""\
-              % (self.peer_user, self.peer_ip)
-        if process.system(cmd, shell=True, ignore_status=True):
+        cmd = "pkill iperf; rm -rf /tmp/iperf-master"
+        try:
+            self.run_command(cmd)
+        except CommandFailed as cf:
```
Comment (narasimhan-v) [inline io/net/iperf_test.py]: fail using exitcode instead of exception.

**Comment by smruti77** [comment]
JOB ID     : d20badaf153f5f4a8627a804cf97dfd6f40dc87b
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-07-25T10.35-d20bada/job.log
 (1/1) iperf_test.py:Iperf.test;run-0ff0: FAIL: FAIL: Throughput Actual - 1%, Expected - 90%,                               Throughput Actual value - 11Mb/sec  (48.60 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 48.71 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-07-25T10.35-d20bada/results.html

JOB ID     : 98d743796e8ffc42ffdc9dd4664e2de0fc6fabf1
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-07-25T10.36-98d7437/job.log
 (1/1) uperf_test.py:Uperf.test;run-0f7f: PASS (62.49 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 62.58 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-07-25T10.36-98d7437/results.html

JOB ID     : b84fed3e08dab5ce5e672398665fe2d0829cb8b8
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-07-25T10.42-b84fed3/job.log
 (1/5) netperf_test.py:Netperf.test;generic-f71b: WARN: Test passed but there were warnings during execution. Check the log for details. (631.09 s)
 (2/5) netperf_test.py:Netperf.test;tcp_stream-838c: PASS (41.75 s)
 (3/5) netperf_test.py:Netperf.test;udp_stream-676c: PASS (40.99 s)
 (4/5) netperf_test.py:Netperf.test;tcp_rr-e9a3: PASS (630.98 s)
 (5/5) netperf_test.py:Netperf.test;udp_rr-f69a: PASS (391.66 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 1 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1736.78 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-07-25T10.42-b84fed3/results.html

JOB ID     : 1e2adff08953fbd555304d29bfaf3ade897f3419
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-07-25T12.04-1e2adff/job.log
 (1/1) multicast.py:ReceiveMulticastTest.test_multicast;Parameters-fd9a: PASS (121.93 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 122.03 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-07-25T12.04-1e2adff/results.html

JOB ID     : b32657d91145114f2bc82db23bf7abe87901a03b
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-07-25T12.39-b32657d/job.log
 (1/3) bonding.py:Bonding.test_setup;round-robin-8e64: PASS (7.93 s)
 (2/3) bonding.py:Bonding.test_run;round-robin-8e64: PASS (30.78 s)
 (3/3) bonding.py:Bonding.test_cleanup;round-robin-8e64: WARN: Test passed but there were warnings during execution. Check the log for details. (609.63 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 1 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 648.54 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-07-25T12.39-b32657d/results.html

**Comment by narasimhan-v** [comment]
@vaishnavibhat @abdhaleegit review please

**Pull request merged**